File: 495-teemo-attacking/teemo-attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        output = 0
        curr = 0

        for each in timeSeries:
            if each > curr:
                output += duration
            else:
                output += each + duration - curr
            curr = each + duration
            
        return output


        

File: 495-teemo-attacking/teemo-attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int output = 0;
        int curr = 0;

        for(int each : timeSeries){
            if (each > curr)
            output += duration;
            else
            output += each + duration - curr;

            curr = each + duration;
        }
        return output;
    }
}